listvalidator/no_null_values.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package listvalidator
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var _ validator.List = noNullValuesValidator{}
+var _ function.ListParameterValidator = noNullValuesValidator{}
+
+type noNullValuesValidator struct{}
+
+func (v noNullValuesValidator) Description(_ context.Context) string {
+	return "null values are not permitted"
```

Comment:
nit: This matches some of the language in our "at least"/"required together"/etc. validators
```suggestion
	return "All values in the list must be configured"
```